doc/doxygen/riot.doxyfile

```diff
@@ -818,13 +818,10 @@ EXCLUDE_SYMLINKS       = NO
 # exclude all test directories for example use the pattern */test/*
 
 EXCLUDE_PATTERNS       = */board/*/tools/* \
-                         */cpu/*/include/component/* \
```

Comment:
The only directory that matched this pattern was the one in cpu/samd21 (now cpu/samr21), same as the pio and instance directory, I moved them to a new subdirectory cpu/sam*/include/cmsis. I found that adding a new subdirectory made it more obvious that this is part of Atmel CMSIS.